pytket/extensions/cutensornet/tnstate/simulation.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2019-2023 Quantinuum
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+##
+#     http://www.apache.org/licenses/LICENSE-2.0
+##
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+from enum import Enum
+
+from random import choice  # type: ignore
+from collections import defaultdict  # type: ignore
+import numpy as np  # type: ignore
+
+from networkx import Graph, community  # type: ignore
+
+from pytket.circuit import Circuit, Command, Qubit
+from pytket.transform import Transform
+from pytket.architecture import Architecture
+from pytket.passes import DefaultMappingPass
+from pytket.predicates import CompilationUnit
+
+from pytket.extensions.cutensornet.general import set_logger
+from .general import CuTensorNetHandle, Config, TNState
+from .mps_gate import MPSxGate
+from .mps_mpo import MPSxMPO
+from .ttn_gate import TTNxGate
+
+
+class ContractionAlg(Enum):

```

Comment:
I think I'll change the name of this enum to `SimulationAlgorithm`. Sounds good?

Reply:
Of course!

Reply:
Done (appearing in next commit)